Hi Sergio,
just tried the make file in ..examples/input

There are some errors:

build//home/christoph/Entwicklung/170721_ArduinoOnPc/sergiobd007/ArduinoOnPc/src/cores/arduino/SerialConsole.o: In Funktion `std::thread::thread<void (&)()>(void (&)())':
SerialConsole.cpp:(.text._ZNSt6threadC2IRFvvEJEEEOT_DpOT0_[_ZNSt6threadC5IRFvvEJEEEOT_DpOT0_]+0x7d): Nicht definierter Verweis auf `pthread_create'
/usr/bin/ld generated: Nicht definierter Verweis auf `pthread_create'
collect2: error: ld returned 1 exit status
../../makeNativeArduino.mk:56: die Regel für Ziel „main“ scheiterte
make: *** [main] Fehler 1

The problem could probably be in the make file:
https://stackoverflow.com/questions/1662909/undefined-reference-to-pthread-create-in-linux

--- a/examples/input/Makefile
+++ b/examples/input/Makefile
@@ -1,8 +1,5 @@
 CUSTOM_LIBS := libraries
 
-ARDUINO_LIBS += Adafruit_GFX
-ARDUINO_LIBS += SPI
-
 BUILD_ROOT := build
 include ../../makeNativeArduino.mk
 

--- a/examples/input/main.ino
+++ b/examples/input/main.ino
@@ -1,27 +1,20 @@
 char receivedChar;
-boolean newData = false;
-
-void recvOneChar() {
- if (Serial.available() > 0) {
- receivedChar = Serial.read();
- newData = true;
- }
-}
-
-void showNewData() {
- if (newData == true) {
- Serial.print("This just in ... ");
- Serial.println(receivedChar);
- newData = false;
- }
-}
 
 void setup() {
- Serial.begin(9600);
- Serial.println("<Arduino is ready>");
+    Serial.begin(9600);
+    Serial.println("<Arduino is ready>");
 }
 
 void loop() {
- recvOneChar();
- showNewData();
+    if (Serial.available() > 0) {
+        receivedChar = Serial.read();
+        Serial.print("Received:[");
+        if(receivedChar!='\n')
+            Serial.print(receivedChar);
+        else
+            Serial.print("\\n");
+        Serial.println("]");
+    }
+    Serial.println("Waiting for input");
+    delay(1000);
 }

